tiny_dnn/core/framework/tensor_storage.h

```diff
@@ -68,6 +68,20 @@ static inline size_t product(Container &c) {
                            std::multiplies<size_t>());
 }
 
+template <typename C1, typename C2>
+static inline size_t compute_offset(const C1 &c1, const C2 &c2) {
```

Comment:
Even simpler, you don't even need to do it in reverse order:

`template <typename C1, typename C2> 
static inline size_t compute_offset2(const C1 &c1, const C2 &c2) { 
    //assert(end(c1)-begin(c1) == end(c2)-begin(c2));
    size_t res = 0;
	for (size_t i = 0; i < c2.size(); ++i) { 
	    res *= c2[i];
	    res += c1[i];
	}
	return res;
}`